infra/provider.tf

```diff
@@ -17,7 +17,7 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = "<= 4.74, != 4.75.0"
+      version = "~> 5.5"
```

Comment:
The `~>` (squiggly arrow) of the version constraint(s) only allow bumps to the right-most number — so this will allow `5.6` but not `6.0`.